core/src/main/kotlin/knote/NotebookManagerImpl.kt

```diff
@@ -6,135 +6,99 @@ import knote.api.PageManager
-
-    override fun evalNotebooks() {
-        logger.info("listNotebookFiles: $listNotebookFiles")
-
-        listNotebookFiles
-            .map { it.name.substringBeforeLast(".notebook.kts") }
-            .filter { id ->
-                notebookFilter?.let { filter -> id in filter } ?: true
-            }
-            .forEach {
-                NotebookManagerImpl.compileNotebook(it)
-            }
+    init {
         startWatcher()
+        compileNotebook()
     }
 
     /***
      * evaluates a single notebook file
      * and registers its pages
      */
-    override fun compileNotebook(notebookId: String): Notebook? {
-        if (notebookFilter?.let { notebookId !in it } == true) {
-            logger.error("$notebookId rejected by notebookFilter: $notebookFilter")
-            return null
-        }
+    override fun compileNotebook(): Notebook? {
+        val notebookId = KNote.notebookId
         logger.debug("attempting to construct notebook '$notebookId'")
-        val file = notebooksDir.resolve("$notebookId.notebook.kts")
-        require(file.exists()) {
-            "notebook: $notebookId does not exist ($file)"
+        val file = KNote.notebookDir.resolve("$notebookId.notebook.kts")
+        if(!file.exists()) {
+            logger.error("notebook: $notebookId does not exist ($file)")
+            return null
         }
 //        val id = file.name.substringBeforeLast(".notebook.kts")
         val (notebookScript, reports) = EvalScript.evalScript<NotebookScript>(
             host,
             file,
-            args = *arrayOf(notebookId, workingDir)
+            args = *arrayOf(notebookId, KNote.notebookDir)
         )
-        val notebook = notebooks.getOrPut(notebookId) {
-            NotebookImpl(
+        if(notebook == null) {
+            notebook = NotebookImpl(
                 id = notebookId,
                 file = file
             )
-        } as NotebookImpl
+        }
+        val notebook = notebook as NotebookImpl
         notebook.reports = reports
         notebook.compiledScript = notebookScript
         if (notebookScript == null) {
             logger.error("evaluation failed for notebook $notebookId")
             return notebook
         }
 
-        notebook.pageManager = PageManagerImpl(notebook, host, workingDir)
+        notebook.pageManager = PageManagerImpl(notebook, host)
         return notebook
     }
 
-    override fun getPageManager(notebookId: String): PageManager? {
-        val notebook = compileNotebookCached(notebookId) as? NotebookImpl ?: run {
+    override fun getPageManager(): PageManager? {
+        val notebookId = KNote.notebookId
+        val notebook = compileNotebookCached() as? NotebookImpl ?: run {
             logger.error("cannot load notebook $notebookId")
             return null
         }
         return notebook.pageManager
-            ?: return PageManagerImpl(notebook, host, workingDir).also {
+            ?: return PageManagerImpl(notebook, host).also {
```

Comment:
I'm not 100% sure but I think you can remove the 2nd return since `also` gives you back the object of itself modified